spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/reactive/context/ReactiveWebServerApplicationContext.java

```diff
@@ -66,9 +66,15 @@ public final void refresh() throws BeansException, IllegalStateException {
 			super.refresh();
 		}
 		catch (RuntimeException ex) {
-			WebServerManager serverManager = this.serverManager;
-			if (serverManager != null) {
-				serverManager.getWebServer().stop();
+			try {
+				WebServer webServer = getWebServer();
+				if (webServer != null) {
+					webServer.stop();
+					webServer.destroy();
```

Comment:
Doesn't it make sense to destroy webServer even if `stop()` throws an exception?

_Probably out of the scope for this fix, but it looks like Juergen was thinking about calling `destroy()` method in context of a bean lifecycle: https://github.com/spring-projects/spring-boot/issues/34955#issuecomment-1525344836_

Thank you for the quick action in regard #44101! 🙏

Reply:
I don’t think so. If `stop()` throws an exception, it likely means that some resources, such as files, networks, etc. are still in use, which could cause `destroy()` to fail as well.